scylla-cdc/src/stream_generations.rs

```diff
@@ -0,0 +1,540 @@
+        .await?
+        .rows
+        .unwrap()
+        .into_typed::<(i64,)>();
+
+    // Query returns a number of peers in a cluster, so we need to add 1 to count current node.
+    Ok(rows.next().unwrap().unwrap().0 as usize + 1)
+}
+
+// Choose appropriate consistency level depending on the cluster size.
+async fn select_consistency(session: &Session, query: &mut Query) -> Result<(), Box<dyn Error>> {
+    query.set_consistency(match get_cluster_size(session).await? {
+        1 => Consistency::One,
+        _ => Consistency::Quorum,
+    });
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use scylla::statement::Consistency;
+    use scylla::SessionBuilder;
+    use uuid::v1::{Context, Timestamp};
+    use uuid::Uuid;
+
+    const TEST_STREAM_TABLE: &str = "Test.cdc_streams_descriptions_v2";
+    const TEST_GENERATION_TABLE: &str = "Test.cdc_generation_timestamps";
+    const TEST_KEYSPACE: &str = "Test";
+    const GENERATION_NEW_MILLISECONDS: i64 = 1635882326384;
+    const GENERATION_OLD_MILLISECONDS: i64 = 1635882224341;
+    const TEST_STREAM_1: &str = "0x7fb9f781956cea08c651295720000001";
+    const TEST_STREAM_2: &str = "0x7fc0000000000000c298b9f168000001";
+
+    impl GenerationFetcher {
+        // Constructor intended for testing purposes.
+        fn test_new(session: &Arc<Session>) -> GenerationFetcher {
+            GenerationFetcher {
+                streams_table_name: TEST_STREAM_TABLE.to_string(),
+                generations_table_name: TEST_GENERATION_TABLE.to_string(),
+                session: Arc::clone(session),
+            }
+        }
+    }
+
+    // Constructs mock table with the same schema as the original one's.
+    fn construct_generation_table_query() -> String {
+        format!(
+            "CREATE TABLE IF NOT EXISTS {}(
+    key text,
+    time timestamp,
+    expired timestamp,
+    PRIMARY KEY (key, time)
+) WITH CLUSTERING ORDER BY (time DESC)
+    AND bloom_filter_fp_chance = 0.01
+    AND caching = {{'keys': 'ALL', 'rows_per_partition': 'ALL'}}
+    AND comment = ''
+    AND compaction = {{'class': 'SizeTieredCompactionStrategy'}}
+    AND compression = {{'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}}
+    AND crc_check_chance = 1.0
+    AND dclocal_read_repair_chance = 0.0
+    AND default_time_to_live = 0
+    AND gc_grace_seconds = 864000
+    AND max_index_interval = 2048
+    AND memtable_flush_period_in_ms = 0
+    AND min_index_interval = 128
+    AND read_repair_chance = 0.0
+    AND speculative_retry = '99.0PERCENTILE';",
+            TEST_GENERATION_TABLE
+        )
+    }
+
+    // Constructs mock table with the same schema as the original one's.
+    fn construct_stream_table_query() -> String {
+        format!(
+            "CREATE TABLE IF NOT EXISTS {} (
+    time timestamp,
+    range_end bigint,
+    streams frozen<set<blob>>,
+    PRIMARY KEY (time, range_end)
+) WITH CLUSTERING ORDER BY (range_end ASC)
```

Comment:
every setting below this line is irrelevant, the test would be nicer if the schema definition was as short as possible